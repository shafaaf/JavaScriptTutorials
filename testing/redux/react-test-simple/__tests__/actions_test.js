import configureMockStore from 'redux-mock-store';

// In a real application we would import this from our actions
const createSuccess = todo => ({
  type: 'CREATE_SUCCESS',
  todo,
});

// Create a mock store
const mockStore = configureMockStore();
const store = mockStore({});
describe('action creators', () => {
  it('creates CREATE_SUCCESS when creating a to-do was successful', () => {
    // Dispatch the createSuccess action with the values of a new to-do.
    store.dispatch(createSuccess(
      {
        id: 1,
        title: 'Example',
        project: 'Testing',
        createdAt: '2017-03-02T23:04:38.003Z',
        modifiedAt: '2017-03-22T16:44:29.034Z',
      },
    ));
    expect(store.getActions()).toMatchSnapshot();
  });
});
