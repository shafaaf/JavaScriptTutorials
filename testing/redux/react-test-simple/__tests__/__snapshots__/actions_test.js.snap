// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action creators creates CREATE_SUCCESS when creating a to-do was successful 1`] = `
Array [
  Object {
    "todo": Object {
      "createdAt": "2017-03-02T23:04:38.003Z",
      "id": 1,
      "modifiedAt": "2017-03-22T16:44:29.034Z",
      "project": "Testing",
      "title": "Example",
    },
    "type": "CREATE_SUCCESS",
  },
]
`;
